File: karuta-tracker/src/main/java/com/karuta/matchtracker/entity/Organization.java
package com.karuta.matchtracker.entity;

import jakarta.persistence.*;
import lombok.*;
import com.karuta.matchtracker.util.JstDateTimeUtil;
import java.time.LocalDateTime;

/**
 * 団体エンティティ
 */
@Entity
@Table(name = "organizations")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Organization {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * 団体コード（wasura, hokudai）
     */
    @Column(name = "code", nullable = false, unique = true, length = 50)
    private String code;

    /**
     * 団体名
     */
    @Column(name = "name", nullable = false, length = 200)
    private String name;

    /**
     * テーマカラー（例: #22c55e）
     */
    @Column(name = "color", nullable = false, length = 10)
    private String color;

    /**
     * 締め切りタイプ
     */
    @Enumerated(EnumType.STRING)
    @Column(name = "deadline_type", nullable = false, length = 20)
    private DeadlineType deadlineType;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = JstDateTimeUtil.now();
        updatedAt = JstDateTimeUtil.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = JstDateTimeUtil.now();
    }
}

File: karuta-tracker-ui/src/utils/organization.js
// 団体（organization）表示ユーティリティ

// 既知団体は略称 override、未登録団体は名称の先頭2文字にフォールバックする。
// override を維持するのは、フォールバックだと "北海道大学かるた会"→"北海" のように
// 本番表示が退行するため（例: hokudai は「北大」を保つ）。任意コードの団体でも
// name があれば先頭2文字で表示でき、団体決め打ちを排除できる。
const SHORT_NAME_OVERRIDES = {
  wasura: 'わすら',
  hokudai: '北大',
};

export const getOrgShortName = (org) => {
  if (!org) return '';
  return SHORT_NAME_OVERRIDES[org.code] || (org.name ? org.name.substring(0, 2) : '');
};
